### PR TITLE
Add support for latest version and make it the default version

### LINKED ISSUE #1115: Support tag ":latest" in Public IDs for Package IDs
**Is your feature request related to a problem? Please describe.**

Currently, I have to first search and check the right version number if I want to add a package.

**Describe the solution you'd like**

Support public ids arguments with the `:latest` tag as version number (e.g. `fetchai/oef:latest`), to match with the highest version number in the registry (local or remote).

**Describe alternatives you've considered**
N/A

**Additional context**

This change actually affects many other parts of the framework, but it's better to decompose the feature request in smaller requests, focus on a small part (e.g. `aea add`) and make it backward-compatible when needed.

What are the implications on the registry side? Locally, I think it can be implemented relatively easy, but the registry service must update the APIs in order to accept and process the new tag correctly, I suspect.

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Add support for latest version and make it the default version

## Fixes

Fix #1115 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/test_configurations/test_base.py [hunk 1]
```diff
@@ -427,6 +427,21 @@ def test_public_id_invalid_version():
...
+def test_public_id_from_string():
+    """Test parsing the public id from string."""
+    public_id = PublicId.from_str("author/package:0.1.0")
+    assert public_id.author == "author"
+    assert public_id.name == "package"
+    assert public_id.version == "0.1.0"
+
+
+def test_public_id_from_string_without_version_string():
+    public_id = PublicId.from_str("author/package")
```
Comment (DavidMinarsch) [inline tests/test_configurations/test_base.py:439]: what about allowing the alias `autor/package:latest` as well?

File: aea/registries/base.py [hunk 2]
```diff
@@ -265,19 +345,25 @@ def register(
...
             raise ValueError(
-                "Item already registered with skill id '{}' and name '{}'".format(
-                    skill_id, item_name
-                )
+                f"Item already registered with skill id '{skill_id}' and name '{item_name}'"
             )
-        self._items.setdefault(skill_id, {})[item_name] = item
+
+        if skill_items is not None:
+            self._items.unregister(skill_id)
```
Comment (DavidMinarsch) [inline aea/registries/base.py:361]: Why did you add this?

File: aea/registries/base.py [hunk 3]
```diff
@@ -303,15 +389,18 @@ def _unregister_from_main_index(
...
                 "No item registered with component id '{}'".format(item_id)
             )
         self.logger.debug("Unregistering item with id {}".format(item_id))
         item = name_to_item.pop(item_name)
         if len(name_to_item) == 0:
-            self._items.pop(skill_id, None)
+            self._items.unregister(skill_id)
+        else:
+            self._items.unregister(skill_id)
+            self._items.register(skill_id, name_to_item)
```
Comment (DavidMinarsch) [inline aea/registries/base.py:403]: why do we unregister and reregister?

File: aea/registries/base.py (hunk 2 above)
Comment (marcofavorito) [reply aea/registries/base.py:361]: because we update it later by registering it again, but only in case we didn't find a record.

File: aea/registries/base.py [hunk 4]
```diff
@@ -418,36 +522,40 @@ def register(
...
-            protocol_id, {}
+        protocol_handlers_by_skill = self._items_by_protocol_and_skill.fetch(
+            protocol_id
         )
-        if skill_id in protocol_handlers_by_skill:
+        if (
+            protocol_handlers_by_skill is not None
+            and skill_id in protocol_handlers_by_skill.ids()
+        ):
             # clean up previous modifications done by super().register
```
Comment (DavidMinarsch) [inline aea/registries/base.py:546]: remove comment?

File: aea/registries/base.py (hunk 3 above)
Comment (marcofavorito) [reply aea/registries/base.py:403]: because the registry APIs don't have a method to `update` an item.
Also, we could rely on the side effect of the  `name_to_item` dictionary, but the APIs designed don't guarantee that strategy works.

File: tests/test_configurations/test_base.py (hunk 1 above)
Comment (marcofavorito) [reply tests/test_configurations/test_base.py:439]: we do

File: aea/registries/base.py (hunk 4 above)
Comment (marcofavorito) [reply aea/registries/base.py]: done

File: aea/configurations/base.py [hunk 5]
```diff
@@ -1569,7 +1569,7 @@ def json(self) -> Dict:
         if self.skill_exception_policy is not None:
             config["skill_exception_policy"] = self.skill_exception_policy
         if self.connection_exception_policy is not None:
-            config["connection_exception_policy"] = self.skill_exception_policy
+            config["connection_exception_policy"] = self.connection_exception_policy
```
Comment (marcofavorito) [inline aea/configurations/base.py:1572]: this line was not covered, and it contained a bug

**Comment by marcofavorito** [comment]
needs to be tested with new registry

**Review by DavidMinarsch — APPROVED** [review]
LGTM, let's wait with merging until tested against remote

**Comment by DavidMinarsch** [comment]
I also notice multiple coverage gaps `aea/registries/base.py                                           221     12    95%   137, 141, 147, 165, 301, 447-454, 637`

**Review by DavidMinarsch — APPROVED** [review]
LGTM
